examples/sycamore/parse_qflex.py

```diff
@@ -0,0 +1,119 @@
+# This is a utility script which parses qflex input circuit 
+# into XASM kernel. 
+# Note: the qubit index mapping (lattice to linear) is
+# specific to the Sycamore device.
+import os, re
+
+qubitIdxMap = {}
+qubitIdxMap[5] = 0
+qubitIdxMap[6] = 1
+qubitIdxMap[14] = 2
+qubitIdxMap[15] = 3
+qubitIdxMap[16] = 4
+qubitIdxMap[17] = 5
+qubitIdxMap[24] = 6
+qubitIdxMap[25] = 7
+qubitIdxMap[26] = 8
+qubitIdxMap[27] = 9
+qubitIdxMap[28] = 10
+qubitIdxMap[32] = 11
+qubitIdxMap[33] = 12
+qubitIdxMap[34] = 13
+qubitIdxMap[35] = 14
+qubitIdxMap[36] = 15
+qubitIdxMap[37] = 16
+qubitIdxMap[38] = 17
+qubitIdxMap[39] = 18
+qubitIdxMap[41] = 19
+qubitIdxMap[42] = 20
+qubitIdxMap[43] = 21
+qubitIdxMap[44] = 22
+qubitIdxMap[45] = 23
+qubitIdxMap[46] = 24
+qubitIdxMap[47] = 25
+qubitIdxMap[48] = 26
+qubitIdxMap[49] = 27
+qubitIdxMap[50] = 28
+qubitIdxMap[51] = 29
+qubitIdxMap[52] = 30
+qubitIdxMap[53] = 31
+qubitIdxMap[54] = 32
+qubitIdxMap[55] = 33
+qubitIdxMap[56] = 34
+qubitIdxMap[57] = 35
+qubitIdxMap[58] = 36
+qubitIdxMap[61] = 37
+qubitIdxMap[62] = 38
+qubitIdxMap[63] = 39
+qubitIdxMap[64] = 40
+qubitIdxMap[65] = 41
+qubitIdxMap[66] = 42
+qubitIdxMap[67] = 43
+qubitIdxMap[72] = 44
+qubitIdxMap[73] = 45
+qubitIdxMap[74] = 46
+qubitIdxMap[75] = 47
+qubitIdxMap[76] = 48
+qubitIdxMap[83] = 49
+qubitIdxMap[84] = 50
+qubitIdxMap[85] = 51
+qubitIdxMap[94] = 52
+
+# Change current dir to this file
+os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+def parseFile(fileName):
+    xasmSrcLines = []
+    xasmSrcLines.append('__qpu__ void sycamoreCirc(qbit q) {')
+
+    with open('resources/' + fileName, 'r') as sourceFile:
+        line = sourceFile.readline()
+        while line:
+            line = line.replace('rz', 'Rz')
+            line = line.replace(', ', ',')
+            components = re.split('\s+', line)
+            if len(components) == 4:
+                # 1-q gate
+                gateName = components[1]
+                if gateName.startswith('Rz') is True:
+                    angle = gateName[3:-1]
+                    gateName = 'Rz'
+                if gateName == 'x_1_2':
+                    angle = '1.5708'
+                    gateName = 'Rx'
+                if gateName == 'y_1_2':
+                    angle = '1.5708'
+                    gateName = 'Ry'
+                # This is not correct
```

Comment:
do we need a fix here? says this is not correct

Reply:
Good catch! I was meant to do the linear algebra calculation to figure out exactly what XACC gate it is (I had the matrix form of that specialized qFlex/Cirq gate).

For this estimation, it doesn't affect the result since both H and hz_1_2 gates are dense single-qubit gates. I'll fix them accordingly and remove that comment.

Reply:
Fixed in https://github.com/ORNL-QCI/tnqvm/pull/40/commits/0938c3ff47ed1ff45b93cb9cd2203370d13227b6